datadog/dogstatsd/base.py

```diff
@@ -246,6 +246,7 @@ def get_socket(self):
                     self.socket = sock
                 else:
                     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+                    sock.setblocking(0)
```

Comment:
The order is different compared to the block above, where the blocking is set after `connect`. 
I think your order is fine here, I'm more worried about the one above. But they should probably not be different?

Reply:
Indeed, I harmonised both case, I did some test to ensure it does not break anything.